lib/chef/provider/package.rb

```diff
@@ -218,6 +218,28 @@ def action_reconfig
         end
       end
 
+      def action_lock
+        multipackage_api_adapter(@current_resource.package_name, @new_resource.version) do |name, version|
+          if package_locked(name, version) == true
+            Chef::Log.debug("#{new_resource} is already locked")
+          else
+            lock_package(name, version)
+            Chef::Log.info("#{@new_resource} locked")
+          end
```

Comment:
both these actions need to be wrapped by converge_by in order to have output in the formatter and to correctly why-run (and update the resource, and fire notifications).

you should probably follow the pattern in `action_remove` for how to write these methods.

i'm a bit concerned that i followed this pattern for a reason:

```
   if idempotency_check?
    converge_by "doing things to package(s) x, y, z" do
      multipackage_api_adapter(...) do |name, version|
        do_the_thing()
      end
    end
  end
```

the idempotency check certainly has to come around the converge_by() and if you moved the multipackage_api_check outside of that you'd have multiple converge_by()s which might have been what i was trying to avoid when i wrote that pattern?  i'm not certain why, but i fear there was some reason otherwise wrapping the whole thing in the multipackage_api_adapter call would have been much cleaner.